hadoop-tools/hadoop-aws/src/main/java/org/apache/hadoop/fs/s3a/S3AUtils.java

```diff
@@ -135,6 +139,12 @@ public final class S3AUtils {
 
   private static final String BUCKET_PATTERN = FS_S3A_BUCKET_PREFIX + "%s.%s";
 
+  public static final Set<ChecksumAlgorithm> SUPPORTED_CHECKSUM_ALGORITHMS = ImmutableSet.of(
```

Comment:
let's move this into its own class `...s3a.impl.ChecksumSupport`. S3AUtils is too big and a source of cherrypick pain, so we prefer isolating things now